### PR TITLE
Add backend specific config

### LINKED ISSUE #4: References to aws vault
If keyring is a standalone lib, we should remove references to aws vault
```
$ ag aws.?vault
keychain_test.go
155:	return filepath.Join(os.TempDir(), fmt.Sprintf("aws-vault-test-%d.keychain", time.Now().UnixNano()))

kwallet.go
15:	APPID             = "aws-vault"
16:	FOLDER            = "aws-vault"

file.go
19:	if password := os.Getenv("AWS_VAULT_FILE_PASSPHRASE"); password != "" {
53:		dir = filepath.Join(home, "/.awsvault/keys/")

README.md
4:Keyring provides utility functions for and a common interface to a range of secure credential storage services. Originally developed as part of [AWS Vault](https://github.com/99designs/aws-vault), a command line tool for securely managing AWS access from developer workstations.
8:  * [Secret Service](https://github.com/99designs/aws-vault/pull/98)
9:  * [KDE Wallet](https://github.com/99designs/aws-vault/pull/27)
10:  * [Encrypted File](https://github.com/99designs/aws-vault/pull/63)
47:  * Ideally verify that [aws-vault](https://github.com/99designs/aws-vault) works with your changes (optional)

secrets.go
12:const secretsCollection = "awsvault"
79:		return &secretsError{"The awsvault collection does not exist. Please add a key first"}
119:	// pack the secret into the aws-vault item
```

### PR DESCRIPTION
**Pull request by lox**
Introduces Config object, as per consensus in #8. Also closes #4 and #8.

This is a majorly breaking change as it changes the contract on the `Open` method:

```go
func main() {
	kr, err := keyring.Open(keyring.Config{
		KeychainName: "my-keychain",
		AllowedBackends:     keyring.AvailableBackends(),
	})

	v, err := kr.Get("llamas")
	if err != nil {
		log.Fatal(err)
	}

	log.Printf("llamas was %v", v)
}
```

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	// LibSecretCollectionName is the name collection in secret-service
+	LibSecretCollectionName string
+}
+
+func NewConfig() Config {
+	return Config{}
+}
+
+func (cfg Config) WithServiceName(name string) Config {
```
Comment (Vektah) [inline config.go]: I wonder if all of these could just use the same value in config? `KeychainName` is probably fine?

Comment (lox) [reply config.go]: Yeah, not sure really. They are often quite different concepts. With Keychain, you might want a conceptual service of aws-vault, but in the login keychain.

**Comment by lox** [comment]
Thanks @Vektah! I'll wait for @mtibben to have a look and then merge this.

**Comment by lox** [comment]
Honestly, I'm kind of regretting pulling this out of `aws-vault`.

**Pull request merged**